.github/workflows/go.yml

```diff
@@ -0,0 +1,113 @@
+name: Go Workflow
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: ["**"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+          check-latest: true
+
+      - name: Build
+        run: go build -o app cmd/main.go
+
+      - name: Archive executable
+        uses: actions/upload-artifact@v3
+        with:
+          name: app
+          path: ./app
+
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+          check-latest: true
+
+      - name: Format
+        run: |
+          OUT="$(go fmt $(go list ./... | grep -v /vendor/) 2>&1)"
+          if [ -n "$OUT" ]; then
+              echo "The following files are not correctly formatted"
+              echo "${OUT}"
+              exit 1
+          fi
+
+      - name: Test
+        run: go test -v $(go list ./... | grep -v /vendor/)
+
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+          check-latest: true
+
+      - name: Format
+        run: |
+          OUT="$(go fmt $(go list ./... | grep -v /vendor/) 2>&1)"
+          if [ -n "$OUT" ]; then
+              echo "The following files are not correctly formatted"
+              echo "${OUT}"
+              exit 1
+          fi
+
+  docker:
```

Comment:
Can we rename this step to `container`? Looks nicer and we make clear, that we don't depend on Docker here.

Reply:
sure thing